crates/host/src/wasmbus/mod.rs

```diff
@@ -1039,19 +1104,71 @@ impl Lattice {
         }: Command = serde_json::from_slice(payload.as_ref())
             .context("failed to deserialize provider launch command")?;
 
-        debug!(configuration, link_name, provider_ref, "launch provider");
+        debug!(link_name, provider_ref, ?configuration, "launch provider");
 
-        bail!("TODO");
+        let (path, claims) = crate::fetch_provider(&provider_ref, &link_name)
+            .await
+            .context("failed to fetch provider")?;
+        let mut command = process::Command::new(&path);
+
+        // TODO: Publish events
+        match self.providers.write().await.entry(claims.subject.clone()) {
+            hash_map::Entry::Vacant(entry) => {
+                debug!(?path, link_name, "launch a new provider");
```

Comment:
Nit: if I start two NATS providers, one on link name `foo` and the other `bar`, this will print:

```
path=X link_name=foo launch a new provider
path=X link_name=bar launch a new instance of an existing provider
```

IMO there's no reason to emit different logs between the first provider started and subsequent providers; these should be treated as independent